SpokestackTests/SpeechContextTest.swift

```diff
@@ -0,0 +1,233 @@
+        // recognize
+        l.reset()
+        let didRecognizeExpectation = expectation(description: "didRecognizeExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.recognize.")
+        l.asyncExpectation = didRecognizeExpectation
+        c.notifyListener(.recognize)
+        wait(for: [didRecognizeExpectation], timeout: 1)
+        
+        // timeout
+        l.reset()
+        let didTimeoutExpectation = expectation(description: "didTimeoutExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.timeout.")
+        l.asyncExpectation = didTimeoutExpectation
+        c.notifyListener(.timeout)
+        wait(for: [didTimeoutExpectation], timeout: 1)
+
+        // error
+        l.reset()
+        // no error set still sends the event
+        let didErrorErrorExpectation = expectation(description: "didErrorErrorExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.error.")
+        l.asyncExpectation = didErrorErrorExpectation
+        c.notifyListener(.error)
+        wait(for: [didErrorErrorExpectation], timeout: 1)
+        // error set sends event
+        let didErrorExpectation = expectation(description: "didErrorExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.error.")
+        l.asyncExpectation = didErrorExpectation
+        c.notifyListener(.error)
+        wait(for: [didErrorExpectation], timeout: 1)
+        
+        // trace
+        l.reset()
+        let didNotTraceExpectation = expectation(description: "didNotTraceExpectation never fulfills when testNotifyListener calls SpeechContextTestDelegate.trace.")
+        didNotTraceExpectation.isInverted = true
+        l.asyncExpectation = didNotTraceExpectation
+        // no trace property set means no trace event
+        c.notifyListener(.trace)
+        wait(for: [didNotTraceExpectation], timeout: 1)
+        // traceproperty set means trace event
+        let didTraceExpectation = expectation(description: "didTraceExpectation  fulfills when testNotifyListener calls SpeechContextTestDelegate.trace.")
+        l.asyncExpectation = didTraceExpectation
+        c.trace = "hi"
+        c.notifyListener(.trace)
+        wait(for: [didTraceExpectation], timeout: 1)
+    }
+}
+
+class SpeechContextTestDelegate: SpeechEventListener {
+    /// Spy pattern for the system under test.
+    /// asyncExpectation lets the caller's test know when the delegate has been called.
+    var didDidInit: Bool = false
+    var activated: Bool = false
+    var deactivated: Bool = false
+    var didDidStart: Bool = false
+    var didDidStop: Bool = false
+    var asyncExpectation: XCTestExpectation?
+    
+    func reset() {
+        didDidInit = false
+        asyncExpectation = .none
+    }
+    
+    func didRecognize(_ result: SpeechContext) {
+        guard let _ = asyncExpectation else {
+            XCTFail("SpeechPipelineTestDelegate was not setup correctly. Missing XCTExpectation reference")
```

Comment:
Might be a little silly of me to make this sort of comment in a test, but this guard could be its own `validateSetup()` or something method, which would make the rest of the listener method implementations slightly cleaner.

Reply:
This is all copypasta code for me, so I'm not worried about verboseness